### PR TITLE
[SPARK-47366][PYTHON] Add pyspark and dataframe parse_json aliases

### PR DESCRIPTION
**Pull request by gene-db**
### What changes were proposed in this pull request?

Added the `parse_json` function alias for pyspark and dataframe APIs.

### Why are the changes needed?

Improves usability of the `parse_json` function.

### Does this PR introduce _any_ user-facing change?
Before this change, the following would not be possible:
```
df.select(parse_json(df.json)).collect()
```

### How was this patch tested?
Added unit tests and manual testing.

### Was this patch authored or co-authored using generative AI tooling?
no

### CODE REVIEW

File: python/pyspark/sql/functions/builtin.py [hunk 1]
```diff
@@ -15098,6 +15098,38 @@ def from_json(
...
+    Parses a column containing a JSON string into a :class:`VariantType`.
+
+    .. versionadded:: 4.0.0
+
+    Parameters
+    ----------
+    col : :class:`~pyspark.sql.Column` or str
+        a column or column name JSON formatted strings
+
+        .. # noqa
```
Comment (HyukjinKwon) [inline python/pyspark/sql/functions/builtin.py:15115]: I think we don't need this (?)

File: python/pyspark/sql/functions/builtin.py [hunk 2]
```diff
@@ -15098,6 +15098,38 @@ def from_json(
...
+
+    Returns
+    -------
+    :class:`~pyspark.sql.Column`
+        a new column of VariantType.
+
+    Examples
+    --------
+    >>> from pyspark.sql.types import *
```
Comment (HyukjinKwon) [inline python/pyspark/sql/functions/builtin.py:15124]: this too.

File: python/pyspark/sql/functions/builtin.py [hunk 3]
```diff
@@ -15098,6 +15098,38 @@ def from_json(
...
+    >>> df = spark.createDataFrame([ {'json': '''{ "a" : 1 }'''} ])
+    >>> df.select(to_json(parse_json(df.json)).alias("v")).collect()
```
Comment (HyukjinKwon) [inline python/pyspark/sql/functions/builtin.py:15126]: Let's probably use `show()`. instead.

File: sql/core/src/main/scala/org/apache/spark/sql/functions.scala [hunk 4]
```diff
@@ -6594,6 +6594,24 @@ object functions {
...
   }
 
+  /**
+   * Parses a JSON string and constructs a Variant value.
+   *
+   * @param json a JSON string.
+   *
+   * @since 4.0.0
+   */
+  def parse_json(json: String): Column = parse_json(lit(json))
```
Comment (HyukjinKwon) [inline sql/core/src/main/scala/org/apache/spark/sql/functions.scala:6604]: I would remove this String type signature

File: python/pyspark/sql/functions/builtin.py [hunk 5]
```diff
@@ -15098,6 +15098,38 @@ def from_json(
     return _invoke_function("from_json", _to_java_column(col), schema, _options_to_str(options))
 
 
+@_try_remote_functions
+def parse_json(
```
Comment (HyukjinKwon) [inline python/pyspark/sql/functions/builtin.py:15102]: should be listed in `python/docs/source/reference/pyspark.sql/functions.rst`

File: python/pyspark/sql/functions/builtin.py (hunk 3 above)
Comment (zhengruifeng) [inline python/pyspark/sql/functions/builtin.py:15126]: let's do not use alias to also verify the default column name

File: python/pyspark/sql/functions/builtin.py (hunk 5 above)
Comment (gene-db) [reply python/pyspark/sql/functions/builtin.py:15102]: done.

File: python/pyspark/sql/functions/builtin.py (hunk 3 above)
Comment (gene-db) [reply python/pyspark/sql/functions/builtin.py]: Removed alias.

**Review by gene-db — COMMENTED** [review]
@HyukjinKwon @zhengruifeng Thanks! I updated the PR.

**Review by HyukjinKwon — APPROVED** [review]

**Review by zhengruifeng — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
seems like we should add the implementation for Scala side as well.

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**